src/elm/Chain.elm

```diff
@@ -219,8 +229,32 @@ update ctx msg model =
                     ( newModel, Cmd.none )
 
                 Just nodes ->
-                    updateChain ctx nodes newModel
+                    updateChain ctx (goodNodes model.minFinalizedHeight nodes) newModel
+        ReceivedConsensusStatus res ->
+            case res of
+                Ok consensusStatus ->
+                    ( model
+                    , Api.getBlockInfo consensusStatus.currentEraGenesisBlock ReceivedBlockResponse
+                    )
+
+                Err err ->
+                    ( model, Cmd.none )
+        ReceivedBlockResponse blockInfoRes ->
+            case blockInfoRes of
+                Ok (Api.Block blockInfo) ->
+                    ( { model
+                        | minFinalizedHeight = Just blockInfo.blockHeight
+                      }
+                    , Build.getNodeInfo GotNodeInfo
+                    )
 
+                Ok (Api.BlockNotFound _) ->
+                    ( model
+                    , Cmd.none
+                    )
+
+                _ ->
```

Comment:
So this could be done better by splitting the model and msg into a "pre-init" and "normal operation" ones.

But since this is in maintenance mode, I thought the current solution is simpler.